### DISCUSSION TITLE
Program crash when using motor.reset_angle()

### DISCUSSION THREAD

**Discussion by Banzobotic** [pybricks/support#463] [Q&A] [answered]
If I call motor.reset_angle() when the motor has not yet been used it doesn't crash but when the motor has been used and I want to reset its angle it will crash with `OSError: [Errno 1] EPERM`. 

It happens when I call it as part of a function or directly in the program code.  This is using mindstorms ev3 with pybricks v2. I have tried calling motor.stop() before calling motor.reset_angle() but I get the same error now in `File "pybricks/ev3devices.py", line 19, in stop` instead. 

Any ideas about how I can fix this crash?

**Comment by dlech** [comment]
You need to stop the motor before resetting the angle.

**Reply by laurensvalk** [reply]
This is not required.

Stopping is only required to change the settings, like acceleration or PID parameters.

**Comment by laurensvalk** [comment, marked as answer]
> If I call `motor.reset_angle()`

An angle argument is required, such as `motor.reset_angle(0)`.

If you don't give an argument, it tries to reset it to the absolute value. But EV3 motors only have incremental encoders. This gives an "invalid operation error", which codes to `EPERM` in EV3 MicroPython 2.0.

Is the motor being used by something else like a `DriveBase`? Then you can get this error as well. Feel free to share your whole script if you'd like us to have a closer look.

**Reply by Banzobotic** [reply]
I do give an angle argument but before I call the reset I have used drive.turn(), is this why it crashes?

**Reply by laurensvalk** [reply]
Yes, see [this section in the documentation](https://pybricks.com/ev3-micropython/robotics.html). You cannot use a motor individually while the drive base is using it.

However, you *can* use [`DriveBase.reset()`](https://pybricks.com/ev3-micropython/robotics.html#pybricks.robotics.DriveBase.reset) instead. This might do what you want.

It's not really a crash, by the way. Python raises an exception to let you know that something is not used in the right way. You can catch this in order to take action when this happens, using `try` and `except`. Although you probably don't need it in this application.

**Reply by Banzobotic** [reply]
drive.reset() didn't seem to reset the angle of the motors according to motor.angle() however I was able to use drive.stop() before I reset the motors and that seems to have fixed the issue, thanks for your help!

**Reply by laurensvalk** [reply]
> drive.reset() didn't seem to reset the angle of the motors according to motor.angle() 

Indeed, `Drive.reset()` resets the `distance()` and `angle()` traveled by the drive base.

Glad to hear this helped!
